ion/textreader.go

```diff
@@ -285,6 +294,16 @@ func (t *textReader) nextBeforeTypeAnnotations() (bool, error) {
 		t.state = trsBeforeContainer
 		t.valueType = StructType
 		t.value = StructType
+
+		if isIonSymbolTable(t.Annotations()) {
```

Comment:
Should we also check if we're at the top level?

Reply:
Added top level check